src/formElementPlugins/listBuilder/src/ApplicationView.tsx

```diff
@@ -0,0 +1,332 @@
+import React, { useEffect, useState } from 'react'
+import {
+  Button,
+  Modal,
+  Form,
+  Segment,
+  Table,
+  TableHeaderCell,
+  TableCell,
+  Card,
+  Icon,
+  Label,
+} from 'semantic-ui-react'
+import { ApplicationViewProps } from '../../types'
+import { ResponseFull } from '../../../utils/types'
+import { TemplateElement, TemplateElementCategory } from '../../../utils/generated/graphql'
+import ApplicationViewWrapper from '../../ApplicationViewWrapper'
+import strings from '../constants'
+
+enum DisplayType {
+  CARDS = 'cards',
+  TABLE = 'table',
+}
+
+interface InputResponseField {
+  isValid?: boolean
+  value: ResponseFull
+}
+
+type ListItem = { [code: string]: InputResponseField }
+
+const ApplicationView: React.FC<ApplicationViewProps> = ({
+  element,
+  parameters,
+  onSave,
+  onUpdate,
+  Markdown,
+  initialValue,
+  currentResponse,
+  applicationData,
+  allResponses,
+}) => {
+  const { isEditable } = element
+  const {
+    label,
+    description,
+    createModalButtonText = strings.BUTTON_LAUNCH_MODAL,
+    modalText,
+    addButtonText = strings.BUTTON_ADD,
+    updateButtonText = strings.BUTTON_UPDATE,
+    deleteItemText = strings.BUTTON_DELETE,
+    inputFields,
+    displayFormat = getDefaultDisplayFormat(inputFields),
+    displayType = DisplayType.CARDS,
+  } = parameters
+
+  const [currentInputResponses, setCurrentInputResponses] = useState<ListItem>(
+    resetCurrentResponses(inputFields)
+  )
+
+  const [listItems, setListItems] = useState<ListItem[]>(initialValue?.list ?? [])
+  const [selectedListItem, setSelectedListItem] = useState<number | null>(null)
+  const [open, setOpen] = useState(false)
+  const [inputError, setInputError] = useState(false)
+
+  useEffect(() => {
+    onSave({
+      text: listItems.length > 0 ? createTextString(listItems, inputFields) : undefined,
+      list: listItems,
+    })
+  }, [listItems])
+
+  // This is sent to ApplicationViewWrapper and runs instead of the default responseMutation
+  const innerElementUpdate =
+    (code: string) =>
+    async ({ variables: response }: { variables: InputResponseField }) => {
+      const newResponses = { ...currentInputResponses, [code]: response }
+      setCurrentInputResponses(newResponses)
+      if (!anyErrorItems(newResponses, inputFields)) setInputError(false)
+    }
```

Comment:
to fix the default value (it was race condition between LB3 and 6.

```suggestion
  const innerElementUpdate =
    (code: string) =>
    async ({ variables: response }: { variables: InputResponseField }) => {
      // need to get most recent state of currentInputResponse, thus using callback
      setCurrentInputResponses((currentInputResponses) => {
        const newResponses = { ...currentInputResponses, [code]: response }
        if (!anyErrorItems(newResponses, inputFields)) setInputError(false)
        return newResponses
      })
    }
```